File: src/main/java/com/kkllffaa/meteor_litematica_printer/mixins/ClientPlayerInteractionManagerMixin.java
package com.kkllffaa.meteor_litematica_printer.mixins;

import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import net.minecraft.client.network.ClientPlayerInteractionManager;

@Mixin(ClientPlayerInteractionManager.class)
public class ClientPlayerInteractionManagerMixin {
    @Inject(method = "cancelBlockBreaking", at = @At("HEAD"), cancellable = true)
    private void onCancelBlockBreaking(CallbackInfo info) {
        if (com.kkllffaa.meteor_litematica_printer.Modules.CRUD.AtomicSettings.BreakSettings.breaking) info.cancel();
    }
}

File: src/main/kotlin/com/kkllffaa/meteor_litematica_printer/mixins/ClientPlayerInteractionManagerMixin.kt
package com.kkllffaa.meteor_litematica_printer.mixins

import com.kkllffaa.meteor_litematica_printer.Modules.AtomicSettings.BreakSettings
import net.minecraft.client.network.ClientPlayerInteractionManager
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(ClientPlayerInteractionManager::class)
class ClientPlayerInteractionManagerMixin {
    @Inject(method = ["cancelBlockBreaking"], at = [At("HEAD")], cancellable = true)
    private fun onCancelBlockBreaking(info: CallbackInfo) {
        if (BreakSettings.breaking) info.cancel()
    }
}
